src/lotto/LottoResultPrinter.js

```diff
@@ -0,0 +1,47 @@
+import { MESSAGES } from '../constants.js';
+import { print } from '../utils/index.js';
+
+class LottoResultPrinter {
+  #matchedCountPerMatchOption;
+
+  #price;
+
+  constructor(matchedCountPerMatchOption, price) {
+    this.#matchedCountPerMatchOption = matchedCountPerMatchOption;
+    this.#price = price;
+  }
+
+  printResult() {
+    print(MESSAGES.IO.OUTPUT.WINNING_STATISTICS);
+    this.#printWinningResult();
+    this.#printRateOfReturn();
+  }
+
+  #printWinningResult() {
+    const matchResults = this.#matchedCountPerMatchOption.map(this.#formatMatchResult);
+    print(matchResults.join('\n'));
+  }
+
+  #formatMatchResult({ count, matchedCount, isBonus, prize }) {
+    if (isBonus) {
+      return MESSAGES.IO.OUTPUT.BONUS_MATCH_RESULT(count, prize, matchedCount);
+    }
+
+    return MESSAGES.IO.OUTPUT.MATCH_RESULT(count, prize, matchedCount);
+  }
+
+  #printRateOfReturn() {
+    const totalPrize = this.#matchedCountPerMatchOption.reduce((acc, cur) => {
+      const { matchedCount, prize } = cur;
+
+      return acc + matchedCount * prize;
+    }, 0);
```

Comment:
간단하게 이렇게 적을 수 있을 것 같아요!
```suggestion
    const totalPrize = this.#matchedCountPerMatchOption.reduce((acc, { matchedCount, prize }) => {
      return acc + matchedCount * prize;
    }, 0);
```

Reply:
매개변수에서 바로 구조분해할당 해버리는 방법이 있군요! 중간 변수 둘 필요 없어서 더 나을 것 같아요. 감사합니다:)